### PR TITLE
add servrside mathjax rendering service

### PR DESCRIPTION
**Pull request by ljonka**
using nodejs mathjax-server

### CODE REVIEW

File: MathJax.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
+
+class MathJax extends MathRenderer {
```
Comment (osnard) [inline MathJax.php]: I'd prefer "MathJaxServer", as there already is/was a "MathJax" mode for client side

File: Math.hooks.php [hunk 2]
```diff
@@ -76,6 +76,7 @@ public static function mathModeToString( $mode, $default = 'png' ) {
 			'MW_MATH_PNG'    => 0,
 			'MW_MATH_SOURCE' => 3,
 			'MW_MATH_MATHML' => 5,
+			'MW_MATH_MATHJAX' => 6,
```
Comment (osnard) [inline Math.hooks.php]: I'd suggest to have a new constant and here. Reason see below.

File: MathJax.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+	protected $renderResult = "";
+
+	protected function getMathTableName() {
+		return 'math';
+	}
+
+	public function getHtmlOutput() {
+		return $this->renderResult;
```
Comment (osnard) [inline MathJax.php]: Would it make sense to wrap this into a container?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+```
+git clone https://github.com/ljonka/mediawiki-extensions-Math.git extensions/Math
+echo "wfLoadExtension(\"Math\"); \$wgDefaultUserOptions[\"math\"] = \"mathjax\";" >> LocalSettings.php
+php maintenance/update.php --quick
+```
+
+## Enable MathJax Service
+```
+npm install -g pm2
+cd extensions/Math && npm update && pm2 start server && cd ../../
```
Comment (osnard) [inline README.md]: You assume "process manager 2" here. Is this in an implicit dependency?

File: server.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+var mathjaxServer = require("mathjax-server");
```
Comment (osnard) [inline server.js]: Maybe this file should not be located in the root directory of the extension. A subfolder would be nice.

**Comment by ljonka** [comment]
see last commit with some changes, also svg to png conversion have been added. https://github.com/hallowelt/mediawiki-extensions-Math/pull/1/commits/06761ea924ae303faef44c82b300b7e80f6f3082

File: server.js (hunk 5 above)
Comment (ljonka) [reply server.js]: done

File: README.md (hunk 4 above)
Comment (ljonka) [reply README.md]: no, this is a recomendation to start a nodejs server and run the service as a background job.

File: Math.hooks.php (hunk 2 above)
Comment (ljonka) [reply Math.hooks.php]: done

File: MathJax.php (hunk 3 above)
Comment (ljonka) [reply MathJax.php]: this is related to the base class handling mechanism

File: MathJax.php (hunk 1 above)
Comment (ljonka) [reply MathJax.php]: done, good point.

**Pull request closed without merging**